### PR TITLE
Create admin panel skeleton

### PR DESCRIPTION
**Pull request by ragnarob**

### CODE REVIEW

File: app/utils/useWindowSize.ts [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+    isLgUp: boolean;
+    isXlUp: boolean;
   }>({
     width: undefined,
     height: undefined,
     isMobile: false,
+    isSmUp: false,
+    isMdUp: false,
+    isLgUp: false,
+    isXlUp: false,
```
Comment (ragnarob) [inline app/utils/useWindowSize.ts:25]: They're nice to have when having to react programatically to breakpoints. The tailwind breakpoint sizes will never change so these are safe to use.

**Comment by toastythetoaster** [comment]
[Cloudflare Pages Preview](https://yif-83-admin-skeleton.yiffer-xyz-contributions-2.pages.dev/admin)

File: app/root.tsx [hunk 2]
```diff
@@ -99,17 +99,18 @@ export default function AppWithProviders() {
...
-          px-4 py-1.5 nav-shadowing justify-between mb-4 text-gray-400 w-full"
+        className="flex bg-gradient-to-r from-theme1-primary to-theme2-primary dark:bg-none
+          px-4 py-1.5 nav-shadowing justify-between mb-4 text-gray-400 w-full z-20"
       >
-        <div className="flex items-center justify-between mx-auto flex-grow">
+        <div className="flex items-center justify-between mx-auto flex-grow max-w-full lg:max-w-80p">
           <div className="flex gap-6 items-center">
             <a
-              href="https://yiffer.xyz"
+              href="/"
```
Comment (ErikMelton) [inline app/root.tsx:113]: Is there a reason to use `/` instead of `BASE_URL` ?

File: app/root.tsx [hunk 3]
```diff
@@ -120,7 +121,7 @@ function Layout({ user, children }: { user: UserSession | null; children: React.
               Yiffer.xyz
             </a>
             <a
-              href="https://yiffer.xyz"
+              href="/"
```
Comment (ErikMelton) [inline app/root.tsx:124]: Same comment as above

**Review by ErikMelton — APPROVED** [review]

File: app/routes/admin.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            isIndented
+            href="/admin/tags/new"
+            text="New tag"
+            isSelected={isRoute('tags/new')}
+          />
+          <SidebarLink
+            href="/admin/pending-comics"
+            text="Pending comics"
+            isSelected={isRoute('pending-comics')}
+          />
```
Comment (toastythetoaster) [inline app/routes/admin.tsx:112]: Why not have something like a `SidebarSection` rather than using a prop to indent individual `SidebarLinks`?

**Review by toastythetoaster — APPROVED** [review]

File: app/routes/admin.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  }
+
+  // Close on navigation - aka a link selected
+  useEffect(() => {
+    setIsOpen(alwaysShow);
+  }, [matches]);
+
+  // Prevent initially rendering wrong until the window size has been determined
+  if (delay) {
+    return <></>;
```
[the remark this answers is not in the document]
Comment (ragnarob) [reply app/routes/admin.tsx:49]: I also wanted this but typescript didn't like it. It wants an element!

File: app/routes/admin.tsx (hunk 4 above)
Comment (ragnarob) [reply app/routes/admin.tsx:112]: Is that less work tho? As it stands all that's needed is one single word on two of the components, and a single conditional class string on the div. I'd say that's as easy as it gets - creating extra components would just be overcomplicating things, no?

File: app/routes/admin.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (toastythetoaster) [reply app/routes/admin.tsx:49]: `null` is a valid React component tho.....

**Pull request merged**